File: 8vim/src/main/java/inc/flide/vim8/keyboardactionlisteners/ClipboardActionListener.java
package inc.flide.vim8.keyboardactionlisteners;

import android.view.View;
import inc.flide.vim8.MainInputMethodService;
import java.util.List;


public class ClipboardActionListener extends KeypadActionListener {
    public ClipboardActionListener(MainInputMethodService mainInputMethodService, View view) {
        super(mainInputMethodService, view);
    }

    public List<String> getClipHistory() {
        return mainInputMethodService.getClipboardManagerService().getClipHistory();
    }

    public void onClipSelected(String selectedClip) {
        onText(selectedClip);
    }
}

File: 8vim/src/main/kotlin/inc/flide/vim8/ime/actionlisteners/ClipboardActionListener.kt
package inc.flide.vim8.ime.actionlisteners

import android.view.View
import inc.flide.vim8.Vim8ImeService

class ClipboardActionListener(vim8ImeService: Vim8ImeService, view: View) :
    KeypadActionListener(vim8ImeService, view) {
    val clipHistory: List<String> = emptyList()
//        get() = vim8ImeService.clipboardManagerService.clipHistory

    fun onClipSelected(selectedClip: String) {
        onText(selectedClip)
    }
}
